scipy/optimize/_trustregion_iterative.py

```diff
@@ -0,0 +1,470 @@
+"""Iterative trust-region optimization subproblem."""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.linalg import norm, get_lapack_funcs, solve_triangular, norm
+from ._trustregion import (_minimize_trust_region, BaseQuadraticSubproblem)
+
+__all__ = ['estimate_smallest_singular_value',
+           'singular_leading_submatrix',
+           'IterativeSubproblem']
+
+
+def _minimize_trust_iterative(fun, x0, args=(), jac=None, hess=None,
+                              **trust_region_options):
+    """
+    Minimization of scalar function of one or more variables using
+    a iterative trust-region algorithm.
+
+    Options
+    -------
+    initial_trust_radius : float
+        Initial trust-region radius.
+    max_trust_radius : float
+        Maximum value of the trust-region radius. No steps that are longer
+        than this value will be proposed.
+    eta : float
+        Trust region related acceptance stringency for proposed steps.
+    gtol : float
+        Gradient norm must be less than ``gtol` before successful
+        termination.
+    trust_region_factor, optimization_factor : float
+        Parameters that determines the stop criteria for the subproblem solver.
+        The iterative algoritm for the trust-region subproblem will stop when
+        the computed step ``p`` fullfil
+        ``norm(p)<=trust_region_factor*trust_radius``
+        or when the solution ``m`` for the subproblem fullfil
+        ``m < optimization_factor*m_optimal``, with ``m_optimal`` being the
```

Comment:
@nmayorov

 I was implementing those changes on the code and I realised something: many of the test I implemented to test the IterativeSubproblem class relies on the ability to set ``k_easy``and ``k_hard`` in such way I guarantee that I have a solution that is VERY close to the exact solution to the problem. And then I compare characteristics of the exact solution to the one I have estimated.

The default values I used, that are the ones that work well during optimisation, does not provide a solution that close to the exact one. That way, when  I implemented those parameters as constants, it make much harder for me to test the IterativeSubproblem.

I am thinking in letting ``k_easy``and ``k_hard``as constructor parameters of the class IterativeSubproblem. It would make this class much easier to test and I think would not affect the code legibility.  What do you think about it?